test/cmd/images.sh

```diff
@@ -20,125 +21,126 @@ USE_IMAGES=${USE_IMAGES:-$defaultimage}
-[ "$(oc get is/newrepo --template='{{(index .spec.tags 0).from.kind}}')" == "DockerImage" ]
-oc tag mysql:5.5 newrepo:latest
-[ "$(oc get is/newrepo --template='{{(index .spec.tags 0).from.kind}}')" == "ImageStreamImage" ]
-oc tag mysql:5.5 newrepo:latest --alias
-[ "$(oc get is/newrepo --template='{{(index .spec.tags 0).from.kind}}')" == "ImageStreamTag" ]
+os::cmd::expect_success 'oc tag registry-1.docker.io/openshift/origin:v1.0.4 newrepo:latest'
+os::cmd::expect_success_and_text "oc get is/newrepo --template='{{(index .spec.tags 0).from.kind}}'" 'DockerImage'
+os::cmd::expect_success 'oc tag mysql:5.5 newrepo:latest'
+os::cmd::expect_success_and_text "oc get is/newrepo --template='{{(index .spec.tags 0).from.kind}}'" 'ImageStreamImage'
+os::cmd::expect_success 'oc tag mysql:5.5 newrepo:latest --alias'
+os::cmd::expect_success_and_text "oc get is/newrepo --template='{{(index .spec.tags 0).from.kind}}'" 'ImageStreamTag'
 
 # test creating streams that don't exist
-[ -z "$(oc get imageStreams tagtest3 --template="{{.status.dockerImageRepository}}")" ]
-[ -z "$(oc get imageStreams tagtest4 --template="{{.status.dockerImageRepository}}")" ]
-oc tag mysql:latest tagtest3:latest tagtest4:latest --alias
-[ "$(oc get is/tagtest3 --template='{{(index .spec.tags 0).from.kind}}')" == "ImageStreamTag" ]
-[ "$(oc get is/tagtest4 --template='{{(index .spec.tags 0).from.kind}}')" == "ImageStreamTag" ]
-oc delete is/tagtest is/tagtest2 is/tagtest3 is/tagtest4
-[ "$(oc tag mysql:latest tagtest:new1 --alias | grep "Tag tagtest:new1 set up to track mysql:latest.")" ]
-[ "$(oc tag mysql:latest tagtest:new1 | grep "Tag tagtest:new1 set to mysql@sha256:")" ]
+os::cmd::expect_failure_and_text 'oc get imageStreams tagtest3' 'not found'
+os::cmd::expect_failure_and_text 'oc get imageStreams tagtest4' 'not found'
+os::cmd::expect_success 'oc tag mysql:latest tagtest3:latest tagtest4:latest --alias'
+os::cmd::expect_success_and_text "oc get is/tagtest3 --template='{{(index .spec.tags 0).from.kind}}'" 'ImageStreamTag'
+os::cmd::expect_success_and_text "oc get is/tagtest4 --template='{{(index .spec.tags 0).from.kind}}'" 'ImageStreamTag'
+os::cmd::expect_success 'oc delete is/tagtest is/tagtest2 is/tagtest3 is/tagtest4'
+os::cmd::expect_success_and_text 'oc tag mysql:latest tagtest:new1 --alias' 'Tag tagtest:new1 set up to track mysql:latest.'
+os::cmd::expect_success_and_text 'oc tag mysql:latest tagtest:new1' 'Tag tagtest:new1 set to mysql@sha256:'
 
 # test deleting a spec tag using oc tag
-oc create -f test/fixtures/test-stream.yaml
-[ "$(oc tag test-stream:latest -d | grep "Deleted")" ]
-oc delete is/test-stream
+os::cmd::expect_success 'oc create -f test/fixtures/test-stream.yaml'
+os::cmd::expect_success_and_text 'oc tag test-stream:latest -d' 'Deleted'
+os::cmd::expect_success 'oc delete is/test-stream'
 echo "tag: ok"
 
 # test deleting a tag using oc delete
-os::util::get_object_assert 'is perl' "{{(index .spec.tags 0).name}}" '5.16'
```

Comment:
I refactored to use `os::cmd::expect_success_and_text` as that is a little less opaque - the `oc get` is clearer. I think we can phase out the use of `os::util::get_object_assert`.